### PR TITLE
Feature/libtorch 1.7

### PR DESCRIPTION
**Pull request by junjihashimoto**
- [x] Update libtorch's links
- [x] Fix gen-deps.sh
- [x] Fix codegen to parse new Declarations.yaml
- [x] Add `ArrayRef<double>` type
- [x] Regenerate codes
- [x] Backport of max/min_values 
- [x] Test

### CODE REVIEW

**Comment by tscholak** [comment]
sweet!

**Comment by junjihashimoto** [comment]
I've fixed codes to support libtorch1.7, but some tests do not pass.

**Comment by junjihashimoto** [comment]
https://github.com/hasktorch/hasktorch/pull/481/checks?check_run_id=1380083259
image

File: hasktorch/src/Torch/Typed/Functional.hs [hunk 1]
```diff
@@ -1893,10 +1902,10 @@ allclose rtol atol equalNaN input other =
 -- (Int64,([4,1],[[1],[1],[1],[0]]))
 --
 -- >>> dtype &&& shape &&& (\t' -> D.asValue (toDynamic t') :: [Int]) $ argmax @0 @DropDim t
--- (Int64,([2],[3,1]))
+-- (Int64,([2],[0,1]))
```
Comment (junjihashimoto) [inline hasktorch/src/Torch/Typed/Functional.hs:1905]: t is `fromJust [[0, 1], [-1, 2], [0, 1], [0, -2]] `
So the meaning of  `(Int64,([2],[3,1]))` is the same as `(Int64,([2],[0,1]))`'s one.

Comment (tscholak) [reply hasktorch/src/Torch/Typed/Functional.hs:1905]: thanks, I wonder why they changed the implementation of argmax :)

File: hasktorch/src/Torch/Typed/Functional.hs [hunk 2]
```diff
@@ -1828,34 +1835,36 @@ addmv ::
...
-  Tensor device dtype '[n] ->
-  -- | second input vector
-  Tensor device dtype '[m] ->
-  -- | input tensor
-  Tensor device dtype shape ->
-  -- | output tensor
-  Tensor device dtype shape'
-addr beta alpha vec1 vec2 input = unsafePerformIO $ ATen.cast5 ATen.Managed.addr_tttss input vec1 vec2 beta alpha
+
+-- When libtorch-1.7, addr is deprecated, and following test does not pass, for now the codes is commented out.
```
Comment (tscholak) [inline hasktorch/src/Torch/Typed/Functional.hs:1839]: ok, can you just remove it completely, please?

Comment (junjihashimoto) [reply hasktorch/src/Torch/Typed/Functional.hs:1839]: Yes.

File: hasktorch/src/Torch/Typed/Functional.hs (hunk 1 above)
Comment (junjihashimoto) [reply hasktorch/src/Torch/Typed/Functional.hs:1874]: https://github.com/pytorch/pytorch/pull/37864
Previous libtorch may return last index.

**Review by tscholak — APPROVED** [review]
As always, excellent work @junjihashimoto !

**Pull request merged**